numpy/core/src/npysort/quicksort.c.src

```diff
@@ -198,6 +208,7 @@ stack_pop:
         }
         pr = *(--sptr);
         pl = *(--sptr);
+        cdepth = *(--psdepth);
```

Comment:
Yeah, the goto goes back instead of up. I don't think either is crystalline clear, having the stack the way we do breaks the obviousness of the doubly recursive pseudo code. Even pushing the largest partition is a bit subtle.